src/libcore/fmt/num.rs

```diff
@@ -256,6 +257,168 @@ macro_rules! impl_Display {
     };
 }
 
+// precision() value of formatter that would not change this number
+// works only with non-negative arguments
+// assumes its argument has no trailing zeroes
+// folows the same rules as formatter precision
+macro_rules! default_precision {
+    ($num:expr) => {{
+        let mut tmp = $num;
+        let mut count = 0_usize;
+        while tmp >= 10 {
+            tmp /= 10;
+            count += 1;
+        }
+        count
+    }};
+}
+
+macro_rules! impl_Exp {
+    ($($t:ident),* as $u:ident via $conv_fn:ident named $name:ident) => {
+        fn $name(
```

Comment:
Before digging too deeply into the code itself can you offer any background on this implementation? Whether it's based on any common approaches used elsewhere?

Reply:
Yeah, I'll inline it. It doesn't seem very reusable as-is. I thought breaking it out might improve readability.

Reply:
It is heavily based on `impl_Display!` from the same file. It also makes use of some stuff from float's implementation (for precision).